compute/cloud-client/src/main/java/compute/spots/CreateSpotVm.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute.spots;
+
+// [START compute_spot_create]
+
+import com.google.cloud.compute.v1.AccessConfig;
+import com.google.cloud.compute.v1.Address;
+import com.google.cloud.compute.v1.AttachedDisk;
+import com.google.cloud.compute.v1.AttachedDiskInitializeParams;
+import com.google.cloud.compute.v1.ImagesClient;
+import com.google.cloud.compute.v1.InsertInstanceRequest;
+import com.google.cloud.compute.v1.Instance;
+import com.google.cloud.compute.v1.InstancesClient;
+import com.google.cloud.compute.v1.NetworkInterface;
+import com.google.cloud.compute.v1.Scheduling;
+import java.io.IOException;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CreateSpotVm {
+  public static void main(String[] args)
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Google Cloud project you want to use.
+    String projectId = "your-project-id";
+    // Name of the virtual machine to check.
+    String instanceName = "your-instance-name";
+    // Name of the zone you want to use. For example: "us-west3-b"
+    String zone = "your-zone";
+
+    createSpotInstance(projectId, instanceName, zone);
+  }
+
+  // Create a new Spot VM instance with Debian 11 operating system.
+  public static Instance createSpotInstance(String projectId, String instanceName, String zone)
+          throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    String image;
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    try (ImagesClient imagesClient = ImagesClient.create()) {
+      image = imagesClient.getFromFamily("debian-cloud", "debian-11").getSelfLink();
+    }
+    AttachedDisk attachedDisk = buildAttachedDisk(image, zone);
+    String machineTypes = String.format("zones/%s/machineTypes/%s", zone, "n1-standard-1");
+
+    // Send an instance creation request to the Compute Engine API and wait for it to complete.
+    return createInstance(projectId, zone, instanceName, attachedDisk, true, machineTypes, false);
```

Comment:
Print the response or a message here or in the main method as a signal for completion.